### PR TITLE
Removing Platform and PlatformVersion from NuGetFramework, Adding UAP

### PR DESCRIPTION
**Pull request by emgarten**
1. Removed Platform and PlatformVersion
2. Added the UAP moniker
3. Added UAP compatibility rules
4. Tie breaking for win81/wpa81 scenarios with UAP
5. General clean up for NuGet.Frameworks

//cc @anurse

### CODE REVIEW

**Comment by emgarten** [comment]
//cc @pranavkm

File: src/NuGet.Frameworks/FrameworkNameProvider.cs [hunk 1]
```diff
@@ -40,6 +40,9 @@ public class FrameworkNameProvider : IFrameworkNameProvider
         // subsets, net -> netcore
         private Dictionary<string, HashSet<string>> _subSetFrameworks;
 
+        // framework ordering
+        private List<string> _frameworkPrecedence = new List<string>();
```
Comment (pranavkm) [inline src/NuGet.Frameworks/FrameworkNameProvider.cs]: `readonly`

File: src/NuGet.Frameworks/FrameworkNameProvider.cs [hunk 2]
```diff
@@ -702,5 +714,28 @@ public bool TryGetSubSetFrameworks(string frameworkIdentifier, out IEnumerable<s
             subSetFrameworks = null;
             return false;
         }
+
+        // Order the frameworks if rules exist, this is used to break ties between equally compatible frameworks
+        // EX: UAP -> Win81, WPA81
+        public int CompareFrameworks(NuGetFramework x, NuGetFramework y)
+        {
+            // Using the ordered list of frameworks choose the framework that matches first
+            if (!StringComparer.OrdinalIgnoreCase.Equals(x.Framework, y.Framework))
```
Comment (pranavkm) [inline src/NuGet.Frameworks/FrameworkNameProvider.cs:723]: You could do `string.Equals(x.Framework, y.Framework, StringComparison.OrdinalIgnoreCase)` instead

File: test/NuGet.Frameworks.Test/FrameworkExpanderTests.cs [hunk 3]
```diff
@@ -11,6 +11,42 @@ namespace NuGet.Test
     public class FrameworkExpanderTests
```
Comment (pranavkm) [inline test/NuGet.Frameworks.Test/FrameworkExpanderTests.cs:11]: Could we get unit tests for the comparer?

**Comment by pranavkm** [comment]
:watch:

**Comment by emgarten** [comment]
Updated.

Added tests, readonly

File: test/NuGet.Frameworks.Test/FrameworkComparerTests.cs [hunk 4]
```diff
@@ -9,6 +11,85 @@ namespace NuGet.Test
...
+                fw4,
+            };
+
+            var comparer = new FrameworkPrecedenceSorter(DefaultFrameworkNameProvider.Instance);
+
+            // Act
+            list.Sort(comparer);
+
+            // Assert
+            Assert.Equal(fw1, list[0]);
```
Comment (pranavkm) [inline test/NuGet.Frameworks.Test/FrameworkComparerTests.cs:85]:
``` C#
Assert.Collection(list,
   item => Assert.Same(fw2, item),
   item => Assert.Same(fw3, item),
   item => Assert.Same(fw4, item),
   item => Assert.Same(fw5, item));
```

**Comment by pranavkm** [comment]
:shipit:

File: src/NuGet.Frameworks/FrameworkNameProvider.cs (hunk 2 above)
Comment (analogrelay) [inline src/NuGet.Frameworks/FrameworkNameProvider.cs:723]: Agree. It's probably identical in terms of behavior, but it's a little more "idiomatic" to do it the way Pranav suggested. Either way, not a major thing for this review.

**Pull request closed without merging**